packages/generator/src/edmx-parser/edmx-file-reader.ts

```diff
@@ -57,19 +57,65 @@ function getODataVersion(edmx): ODataVersion {
 function getRoot(edmx) {
   const schema = edmx['edmx:Edmx']['edmx:DataServices'].Schema;
   if (schema.length > 1) {
-    if (schema.length > 2) {
-      throw new Error(
-        'There are more than two schemas in the input metadata file.'
-      );
-    }
-    // We assume SFSF edmx files to always have multiple schema tags
-    logger.info(`${schema.length} schemas found. Schemas will be merged.`);
-    return schema.reduce(
-      (mergedSchemas, schemaEntry) => ({ ...mergedSchemas, ...schemaEntry }),
-      {}
-    );
+    return mergeSchemas(schema);
   }
-  return schema;
+  return mergeSchemas(forceArray(schema));
+}
+
+function mergeSchemas(schema) {
+  return {
+    EntityContainer: flat(
+      schema.map(s =>
+        forceArray(s.EntityContainer).map(ec =>
+          addNamespaceToEntityContainer(ec, s.Namespace)
+        )
+      )
+    ),
+    EntityType: flat(
+      schema.map(s => forceArray(s.EntityType).map(addNamespace(s.Namespace)))
+    ),
```

Comment:
Looks like code duplication to me ;)

Independently of that, I think we should merge this with the functionality of edmx-parser and move it to parser.